### PR TITLE
Change `mlflow.autolog` metrics name to the same as `mlflow.evaluate`

### PR DESCRIPTION
**Pull request by wenfeiy-db**
## What changes are proposed in this pull request?

Currently the metric names from `autolog` (mse) and `evaluate` (mean_squared_error) are different. This PR change the metric names in `autolog` to the same as ones in `evaluate`

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

The metric names logged by `mlflow.autolog` are now the same as `mlflow.evaluate`.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by harupy** [comment]
@mlflow-automation autoformat

**Review by WeichenXu123 — APPROVED** [review]

**Review by dbczumar — APPROVED** [review]
LGTM!

**Comment by wenfeiy-db** [comment]
@dbczumar The cross version tests are failing because I changed the metric strings in the tests. Why does the new test have to pass with older versions? What's the typical practice here?

**Comment by dbczumar** [comment]
> @dbczumar The cross version tests are failing because I changed the metric strings in the tests. Why does the new test have to pass with older versions? What's the typical practice here?

@wenfeiy-db These tests don't run against older versions of MLflow, they run against older versions of sklearn. This indicates that your changes don't work properly across varying versions of scikit-learn. I suspect that the `models` tests won't pass either.

**Comment by wenfeiy-db** [comment]
> > @dbczumar The cross version tests are failing because I changed the metric strings in the tests. Why does the new test have to pass with older versions? What's the typical practice here?
> 
> @wenfeiy-db These tests don't run against older versions of MLflow, they run against older versions of sklearn. This indicates that your changes don't work properly across varying versions of scikit-learn. I suspect that the `models` tests won't pass either.

Thanks for the explanation! I actually found a bug, and the tests are now passing with sklearn 0.22.1 & 1.11.3 (at least locally).
